lib/modules/dosomething/dosomething_reportback/dosomething_reportback.module

```diff
@@ -690,14 +690,16 @@ function dosomething_reportback_save($values) {
   }
   else {
     $entity = entity_load_single('reportback', $rbid);
+    $reportback_item = entity_load_single('reportback_item', $values['fid']);
   }
   try {
     // Set entity properties.
     dosomething_reportback_set_properties($entity, $values);
     // Set entity files.
-    dosomething_reportback_set_files($entity, $values);
-    // Save the entity.
+    dosomething_reportback_set_files($reportback_item, $values);
```

Comment:
Ohh still might be a little confused - so we want to update the caption for the file in the `else` block right before the `try`?